### ISSUE TITLE
Hardcoded column name

### ISSUE THREAD

**Issue by a-komarev** [cmgmyr/laravel-messenger#74]
In my application there is no `users.name` column but participantsString using it. User names are separated to different table `profile`. As I can see there will be great thing to add an utility method in User model which will return a user name and use it inside your package.

**Comment by a-komarev** [comment]
To clarify situation more, it's not duplicate of #39
I need to get user data from `profiles` table, not from `users`.

**Comment by askaoru** [comment]
not sure if it might work or not. but to give an idea, you might use this instead
````
$thread->getParticipantFromUser(Auth::id())
````
then use database relationship to fetch the username from your profile table.
```
$thread->getParticipantFromUser(Auth::id())->profiles->username
````
sorry, not a pro here and i couldn't test it out right now

**Comment by cmgmyr** [comment]
@a-komarev once #75 is done, I believe you should be all set with this. Once that is completed you should be able to do something like this:

```php
$thread->participantsString($userId, ['username'])
```

since the `name` referenced in the queries are just aliases. Unless I'm missing something else?

**Comment by a-komarev** [comment]
@askaoru 
Thanks for idea! Finally this pretty complex query, but it works
```php
$thread->getParticipantFromUser(Auth::id())->user->profile->username
```

@cmgmyr 
The issue in that case - in my current application there is no any public data in users table. And I need to fetch it from model related to user.

**Comment by cmgmyr** [comment]
ok, makes a lot more sense now. Since this is the first time I'm hearing about something like this I don't think I want to go down the road of trying to map out these relationships within the package. So as long as you have a query that you can use (even though it's a bit ugly), that seems to be a good solution to me. If others run into this same problem down the road I'd be happy to revisit

**Comment by a-komarev** [comment]
Yes, I suppose it's rare to see such separating. Solution provided above suitable for 1 to 1 chat only, because it's getting only one user at time. And final solution looks like this one:

```php
$thread->getParticipantFromUser($thread->participantsString(Auth::id(), ['id']))->user->profile->present()->fullName
```

In current application there is no group chat yet, so it's ok for me. But I will look for it later and try to provide any kind of solutions if this issue will be actual still.
